src/aatoolbox/utils/raster.py

```diff
@@ -0,0 +1,447 @@
+        and the input  will be returned.
+
+        Parameters
+        ----------
+        inplace : bool, optional
+            If True, will overwrite existing data array. Default is False.
+
+        Returns
+        -------
+        xarray.DataArray
+            Dataset with transformed longitude coordinates.
+
+        Examples
+        --------
+        >>> import xarray
+        >>> import numpy
+        >>> da = xarray.DataArray(
+        ...  numpy.arange(16).reshape(4,4),
+        ...  coords={"lat":numpy.array([87, 88, 89, 90]),
+        ...          "lon":numpy.array([5, 120, 199, 360])}
+        ... )
+        >>> da_inv = da.aat.change_longitude_range()
+        >>> da_inv.get_index("lon")
+        Int64Index([-161, 0, 5, 120], dtype='int64', name='lon')
+        """
+        data_obj = self._get_obj(inplace=inplace)
+        lon_min = data_obj.indexes[self.x_dim].min()
+        lon_max = data_obj.indexes[self.x_dim].max()
+
+        if lon_max > 180:
+            logger.info("Converting longitude from 0 360 to -180 to 180.")
+
+            data_obj = data_obj.assign_coords(
+                {self.x_dim: ((data_obj[self.x_dim] + 180) % 360) - 180}
+            ).sortby(self.x_dim)
+
+        elif lon_min < 0:
+            logger.info("Converting longitude from -180 to 180 to 0 to 360.")
+
+            data_obj = data_obj.assign_coords(
+                {
+                    self.x_dim: np.where(  # noqa: FKA01
+                        data_obj[self.x_dim] < 0,
+                        data_obj[self.x_dim] + 360,
+                        data_obj[self.x_dim],
+                    )
+                }
+            ).sortby(self.x_dim)
+
+        else:
+            logger.info(
+                "Indeterminate longitude range and no need to convert."
+            )
+
+        return data_obj if not inplace else None
+
+    def compute_raster_statistics(
+        self,
+        gdf: gpd.GeoDataFrame,
+        feature_col: str,
+        stats_list: List[str] = None,
+        percentile_list: List[int] = None,
+        all_touched: bool = False,
+        geom_col: str = "geometry",
+    ) -> pandas.DataFrame:
+        """Compute raster statistics for polygon geometry.
+
+        ``compute_raster_statistics()`` is designed to
+        quickly compute raster statistics across a polygon
+        and its features.
+
+        Parameters
+        ----------
+        gdf : geopandas.GeoDataFrame
+            GeoDataFrame with row per area for stats computation.
+        feature_col : str
+            Column in ``gdf`` to use as row/feature identifier.
+        stats_list : List[str], optional
+            List of statistics to calculate, by default None.
+            Passed to ``get_attr()``.
+        percentile_list : List[int], optional
+            List of percentiles to compute, by default None.
+        all_touched : bool, optional
+            If ``True`` all cells touching the region will be
+            included, by default False.
```

Comment:
I know I wrote this but would add if ``False`` only cells with their centre in the region will be included